README.md

```diff
@@ -42,13 +42,114 @@ Include OSC in your .csproj file.
 </Project>
 ```
 
+### Connecting
+
+You can connect to your OpenSearch cluster via a single node, or by specifying multiple nodes using a node pool. Using a node pool has a few advantages over a single node, such as load balancing and cluster failover support.
+
 **Connecting to a single node**
 ```csharp
 var node = new Uri("http://myserver:9200");
-var settings = new ConnectionSettings(node);
+var config = new ConnectionConfiguration(node);
+var client = new OpenSearchClient(config);
+```
+
+**Connecting to multiple nodes using a connection pool**
+```csharp
+var nodes = new Uri[]
+{
+	new Uri("http://myserver1:9200"),
+	new Uri("http://myserver2:9200"),
+	new Uri("http://myserver3:9200")
+};
+
+var pool = new StaticNodePool(nodes);
+var settings = new ConnectionSettings(pool);
 var client = new OpenSearchClient(settings);
 ```
 
+### Indexing
+
+Indexing a document is as simple as:
+
+```csharp
+var tweet = new Tweet
+{
+    Id = 1,
+    User = "kimchy",
+    PostDate = new DateTime(2009, 11, 15),
+    Message = "Trying out OSC, so far so good?"
+};
+
+var indexingResponse = client.Index(tweet, idx => idx.Index("mytweetindex")); //or specify index via settings.DefaultIndex("mytweetindex");
+```
+
+All the calls have async variants:
+
+```csharp
+var indexingResponse = client.IndexAsync(tweet, idx => idx.Index("mytweetindex")); // returns a Task<IndexResponse>
+
+// Or, in an async-context
+var indexingResponse = await client.IndexAsync(tweet, idx => idx.Index("mytweetindex")); // awaits a Task<IndexResponse>
+```
+
+### Getting a document
+
+```csharp
+var getResponse = client.Get<Tweet>(indexingResponse.Id, idx => idx.Index("mytweetindex")); // returns an IGetResponse mapped 1-to-1 with the OpenSearch JSON response
+var tweet = getResponse.Source; // the original document
+```
+
+### Searching for documents
+
+OSC exposes a fluent interface and a [powerful query DSL](https://opensearch.org/docs/latest/opensearch/query-dsl/index/)
+
+```csharp
+var searchResponse = client.Search<Tweet>(s => s
+    .Index("mytweetindex") //or specify index via settings.DefaultIndex("mytweetindex");
+    .From(0)
+    .Size(10)
+    .Query(q => q
+        .Term(t => t.User, "kimchy") || q
+        .Match(mq => mq.Field(f => f.User).Query("osc"))
+    )
+);
+```
+
+As well as an object initializer syntax if lambdas aren't your thing:
+
+```csharp
+var request = new SearchRequest
+{
+    From = 0,
+    Size = 10,
+    Query = new TermQuery { Field = "user", Value = "kimchy" } || 
+            new MatchQuery { Field = "description", Query = "OSC" }
+};
+
+var searchResponse = client.Search<Tweet>(request);
+```
+### Falling back to OpenSearch.Net
+
+OSC also includes and exposes the low-level [OpenSearch.Net](https://github.com/opensearch-project/opensearch-net/tree/main/src/OpenSearch.Net) client that you can fall back to in case anything is missing:
+
+```csharp
+//.LowLevel is of type IOpenSearchLowLevelClient
```

Comment:
I like that we mention the type of intermediate properties. 

It'd be more clear to use OpenSearchClient.LowLevel to refer to LowLevel property in the comments.

Same for the Search method -- IOpenSearchLowLevelClient.Search.

Reply:
See update in 162693602.